### ISSUE TITLE
flycheck does not correctly detect warnings from nvcc

### ISSUE THREAD

**Issue by eyal0** [flycheck/flycheck#2099]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist

- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description

flycheck does not correctly detect warnings.

## Steps to reproduce

1. Put this into a file `int main() { size_t x = 5; }`
2. `C-c ! C-c` to cause flycheck compilation and select `cuda-nvcc`
3. Note that a warning appears from the compiler yet the buffer doesn't show any warnings.

The compile buffer looks like this:

```
-*- mode: compilation; default-directory: "~/coding/" -*-
Compilation started at Wed Dec 11 13:33:49

/usr/local/cuda/bin/nvcc -c --output-file /dev/null --x\=cu -rdc\=true -std\=c\+\+20 --expt-relaxed-constexpr --expt-extended-lambda -gencode arch\=compute_70\,code\=sm_70 -I . /home/esoha/coding/foo.cu
/home/esoha/coding/foo.cu(1): warning #177-D: variable "x" was declared but never referenced
  int main() { size_t x = 5; }
                      ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"


Compilation finished at Wed Dec 11 13:33:50, duration 1.45 s
```

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior

I expected a warning to show up in the little bar below the buffer in emacs, in yellow, with a "1".

## Screenshots

![image](https://github.com/user-attachments/assets/3bf1a8f7-4108-47a4-b58e-b5d2073d8dfd)

```
Syntax checkers for buffer foo.cu in cuda-mode:

First checker to run:

  cuda-nvcc
    - may enable: yes
    - executable: Found at /usr/local/cuda/bin/nvcc

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: nil
Emacs version:    30.0.50
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes

I'll take a look at this myself, too.

**Comment by eyal0** [comment]
The PR seems to complete solve the issue.  Thanks for maintaining this wonderful tool!
